### DISCUSSION TITLE
DAI 2024-2025 - Java IOs benchmarking - Gianni Cecchetto

### DISCUSSION THREAD

**Discussion by GianniCecchetto** [heig-vd-dai-course/heig-vd-dai-course#353] [Show and tell]
[The link to your GitHub repository](https://github.com/GianniCecchetto/05-java-ios)  

| File size | BINARY read | BUFFERED_BINARY read | TEXT read | BUFFERED_TEXT read | BINARY write | BUFFERED_BINARY write | TEXT write | BUFFERED_TEXT write |
|-----------|-------------|----------------------|-----------|--------------------|--------------|-----------------------|------------|---------------------|
| 1B        | 165         | 165                  | 165       | 165                | 177          | 169                   | 173        | 168                 |
| 1KiB      | 165         | 165                  | 170       | 170                | 193          | 169                   | 185        | 173                 |
| 1MiB      | 923         | 179                  | 303       | 230                | 2827         | 204                   | 258        | 198                 |
| 5MiB      | 3440        | 250                  | 405       | 267                | 12805        | 225                   | 386        | 242                 |

- Which type of stream is the most efficient for each use case?  
The most efficient for each use case is BUFFERED_BINARY.

- Why is it more efficient than the other types of streams?  
Because it is BUFFERED, and doesn't have the encoding that the TEXT type has.

- What is the difference between binary data and text data?  
Text data has an encoding.

- What is a character encoding?  
It's a way to represent a character by a number.

- Why is this benchmark methodology important?
Because it helps use choose which one is the best for our current utilization.

**Comment by ludelafo** [comment]
Thank you!

You can always find the solution in the solution repository if you ever need to compare your work with the official one here: https://github.com/heig-vd-dai-course/heig-vd-dai-course-solutions.

Do not hesitate to ask questions if something is not clear, either in this discussion or in the Q&A category of the GitHub Discussions here: https://github.com/orgs/heig-vd-dai-course/discussions/categories/q-a.
